File: py/p6.py
#!/usr/bin/env python
#
# Finds the difference between the sum of the squares of the first one hundred
# numbers and the square of the sum.
#
# (c) 2013 Matthew Rheaume

def solve():
	n = 100

	sum_total = 0
	sum_to_n = sum(xrange(1,n+1))

	for i in xrange(1,n+1):
		sum_total += i * (sum_to_n - i)

	return sum_total

if __name__ == "__main__":
	print solve()

File: hs/p6.hs
-- Finds the difference between the sum of the squares of the first one hundred
-- numbers and the square of the sum.
--
-- (c) 2013 Matthew Rheaume

module Main where

squareSumDiff :: Int -> Int
squareSumDiff n = foldr sumProd 0 [1..n] where
	sumToN = sum [1..n]
	sumProd x = (x * (sumToN - x) +)

solve :: Int
solve = squareSumDiff 100

main :: IO ()
main = putStrLn $ show $ solve
